src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -103,24 +105,36 @@ internal sealed class RSession : IRSession, IRCallbacks {
 
         public Task<IRSessionInteraction> BeginInteractionAsync(bool isVisible = true, CancellationToken cancellationToken = default(CancellationToken)) {
             if (!_isHostRunning) {
-                return CanceledInteractionTask;
+                return CanceledBeginInteractionTask;
             }
 
             RSessionRequestSource requestSource = new RSessionRequestSource(isVisible, _contexts, cancellationToken);
             _pendingRequestSources.Post(requestSource);
 
-            return _isHostRunning ? requestSource.CreateRequestTask : CanceledInteractionTask;
+            return _isHostRunning ? requestSource.CreateRequestTask : CanceledBeginInteractionTask;
         }
 
         public Task<IRSessionEvaluation> BeginEvaluationAsync(bool isMutating = true, CancellationToken cancellationToken = default(CancellationToken)) {
             if (!_isHostRunning) {
-                return CanceledEvaluationTask;
+                return CanceledBeginEvaluationTask;
             }
 
             var source = new RSessionEvaluationSource(isMutating, cancellationToken);
             _pendingEvaluationSources.Post(source);
 
-            return _isHostRunning ? source.Task : CanceledEvaluationTask;
+            return _isHostRunning ? source.Task : CanceledBeginEvaluationTask;
+        }
+
+        public async Task<REvaluationResult> EvaluateAsync(string expression, REvaluationKind kind, CancellationToken ct = default(CancellationToken)) {
```

Comment:
It's there to write a `try`/`catch` that converts `MessageTransportException` when host suddenly goes down to cancellation.